cEP-0020.md

```diff
@@ -0,0 +1,167 @@
+
+## Process involved in the Gamification system
+
+As the goal of this project is to make use of metrics and gamification to motivate and support newcomers to overcome the orientation barriers to contribute to the community.
+To achieve this goal we have selected three game design elements to help newcomers: Levels, Points, and Badges. For each of these elements, we will have a set of rules to describe the operation of the gamified environment. i.e.:
+
+- **Levels** group different tasks, enabling users to unlock specific and contextualize set of actions. Levels will be applied to make newcomers focus on a given step before moving ahead on the contribution process. E.g.: We will divide our newcomer process into three levels and each level will contain three tasks:
+
+  - Level-I:
+    - Joining our gitter channel and the community
+    - Running CI tools to your fork
+    - Running coala-ci on a popular github repo
+  - Level-II:
+    - Getting assigned to a difficulty/newcomer issue
+    - Creating and merging PR for that issue
+    - Reviewing at least one difficulty/newcomer issue
+  - Level-III:
+    - Getting assigned to a difficulty/low issue
+    - Creating and merging a PR for that issue
+    - Reviewing at least one difficulty/low issue
+
+- **Points** will be the most applied element out of three game design elements. Acquiring points means that the actions performed by the users are in accordance with what was expected. Points can be earned by means of performing actions to the different tasks under levels. E.g.: If a user wants to go to next level he will have to collect a certain number of points from doing the tasks under levels. But in case of Level-II and Level-III if a user is not able to acquire the required number of points after performing all the tasks, he will have to review more PRs to collect the required points.
+
+- **Badges** enable newcomers to follow their performance and compare to other newcomers.**After completing the level three successfully the newcomer will be a developer**. And he will be awarded a badge based on their performance in each level and the total number of points.
+
+These all are just basic ideas, the process and tasks in the levels can be changed according to the use case and implementaion process.
+
+A similar process can also be extended and applied to developers to achieve more number of badges or something more.
```

Comment:
badges and points among developers sound cool. After a threshold developers can get the ability to atleast `unack` PRs, apart from the issue I already opened about it. (I think it's gonna take time for the gitmate plugins to get out, but the recent comments on that which I have failed to read may be of importance over here). `unack` does much much less harm than an `ack`. A Gitlab like feature would also be beneficial over here, requiring acks by 3 or more developers can automatically assign the PRs a tag, like `approved by developers` or something so that maintainers can clear them out as early as possible. ability to `ack` can be given at a higher threshold.

Reply:
Stretch goal stuff, don't let this get in the way of you completing the project